viscy/representation/evaluation/lca.py

```diff
@@ -56,23 +64,44 @@ def fit_logistic_regression(
     tuple[LogisticRegression, tuple[tuple[NDArray, NDArray], tuple[NDArray, NDArray]]]
         Trained classifier and data split [[X_train, y_train], [X_test, y_test]].
     """
-    fov_selection = features["fov_name"].isin(train_fovs)
-    train_selection = fov_selection
-    test_selection = ~fov_selection
     annotations = annotations.cat.codes.values.copy()
+
+    # Handle background class removal before splitting for stratification
     if remove_background_class:
-        label_selection = annotations != 0
-        train_selection &= label_selection
-        test_selection &= label_selection
-        annotations -= 1
-    train_features = features.values[train_selection]
-    test_features = features.values[test_selection]
+        valid_indices = annotations != 0
+        features_filtered = features[valid_indices]
+        annotations_filtered = annotations[valid_indices] - 1
+    else:
+        features_filtered = features
+        annotations_filtered = annotations
+
+    # Determine train/test split
+    if train_fovs is not None:
+        fov_selection = features_filtered["fov_name"].isin(train_fovs)
+        train_selection = fov_selection
+        test_selection = ~fov_selection
+    else:
+        # Use stratified sampling
+        n_samples = len(annotations_filtered)
+        indices = range(n_samples)
+        train_indices, test_indices = train_test_split(
```

Comment:
Will this split the same track into train and test?

Reply:
at a fov level